ddtrace/contrib/psycopg/async_cursor.py

```diff
@@ -21,6 +21,25 @@ async def __aexit__(self, exc_type, exc_val, exc_tb) -> None:
         # messages will be the same.
         return await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
 
+    async def __aiter__(self):
+        await self._fetch_pipeline()
+        self._check_result_for_fetch()
+
+        def load(pos: int):
+            return self._tx.load_row(pos, self._make_row)
+
+        while True:
+            row = load(self._pos)
+            if row is None:
+                break
+            self._pos += 1
+            yield row
+
+    async def _fetch_pipeline(self) -> None:
+        if self._execmany_returning is not False and not self.pgresult and self._conn._pipeline:
+            async with self._conn.lock:
+                await self._conn.wait(self._conn._pipeline._fetch_gen(flush=True))
```

Comment:
Although we need to implement `__aiter__`, we don't want to re implement the wrapped cursor's functionality. Instead, we want to ensure we are correctly calling the wrapped method from our new `__aiter__` method. 
```suggestion
    def __aiter__(self):
        return self.__wrapped__.__aiter__()
```

Reply:
this change should also probably be moved up into the dbapi async cursor parent class of this